### ISSUE TITLE
User given vDev names do not survive a Zway restart (v2.2.4-rc17)

### ISSUE THREAD

**Issue by pz1** [Z-Wave-Me/zwave-smarthome#187]
_Smarthome_ - After a reboot changed _vDev names_ are reverted to the original name given in _index.js_ in the _overlay section_. Observed it with my own modules under userModules. But it doesn't seem to happen in the included standard modules. I have seen this behaviour before in much older releases. This happens also on my rc16 system with Popp smoke alarm. I had given it a human understandable name, but the element name has also been reverted to the system generated **Smoke Alarm (23.0.48.2)**
**update 20161006** Still a problem in rc20

**Comment by ronie-z-wave** [comment]
Can you send me the name of apps where you could observe that? 
I've tested a POPP smoke sensor without retracing it. 

Could be that this behaviour is caused by overlay property within devices.create() object

**Comment by pz1** [comment]
Thanks for the response. I do not see the behaviour in included devices like the Popp anymore. Don't understand why it did happen before. 
Your last remark triggered me to have a different look at my own Apps.  For example my Wunderground app (not to be confused with the extended one from Maros Kollar) the sensor takes the name from the city, as defined in the configuration form when it is setup. The problem occured because I did change the element name _local weather_ that apparantly is overwritten by the confgured city_name _Ulrum_ at z-way restart:

![ulrum](https://cloud.githubusercontent.com/assets/7747627/19477246/8770def0-953d-11e6-981f-33033b28c8a9.PNG)

This [Wunderground App](https://github.com/pz1/ZWayModules/tree/master/Wunderground) has been derived from, I think an earlier version, of your OpenWeather App

**Comment by ronie-z-wave** [comment]
Jup overlay seems to be the cause:

https://github.com/Z-Wave-Me/home-automation/blob/develop/classes/VirtualDevice.js

line 114 and following

1. first merge current object with latest stored in config.json 
2. apply overlay

from http://underscorejs.org/ :

**defaults** _.defaults(object, *defaults) 
Fill in undefined properties in object with the first value present in the following list of defaults objects.

var iceCream = {flavor: "chocolate"};
_.defaults(iceCream, {flavor: "vanilla", sprinkles: "lots"});
=> {flavor: "chocolate", sprinkles: "lots"}

**extend** _.extend(destination, *sources) 
Shallowly copy all of the properties in the source objects over to the destination object, and return the destination object. Any nested objects or arrays will be copied by reference, not duplicated. It's in-order, so the last source will override properties of the same name in previous arguments.

_.extend({name: 'moe'}, {age: 50});
=> {name: 'moe', age: 50}

**Comment by pz1** [comment]
@ronie-z-wave , I am puzzled about this response. I have read it over and again, but as a non-programmmer I find it difficult to grasp.. 
Is your post meant as an instruction for me to mend my failing modules, or as an analysis of the problem to be mended by you.

**Comment by pz1** [comment]
I don't know if this is related with a recent change I made in my Wunderground module. In [index.js](https://github.com/pz1/ZWayModules/blob/master/Wunderground/index.js) I initially set deviceType:text for vDev2 and 3. While text is a valid deviceType, it did not result in the value being displayed in Smarthome. In my JS sources I changed the type to sensorMultilevel. Still no display of values. Only after I changed this to sensorMultilevel with nano directly in configjson-xxx.json, the values became visible in Smarthome. 
PS: I had to do the same editing on the installation that imported the vDevs
